common/bootstraptoken/token.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstraptoken
+
+import (
+	"fmt"
+	"time"
+
+	infrastructurev1beta1 "github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/apis/infrastructure/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
+	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+)
+
+// GetTokenIDSecretFromBootstrapTokenStr splits the token string and returns the tokenID and tokenSecret parts
+func GetTokenIDSecretFromBootstrapTokenStr(tokenStr string) (tokenID, tokenSecret string, err error) {
+	substrs := bootstraputil.BootstrapTokenRegexp.FindStringSubmatch(tokenStr)
+	if len(substrs) != 3 { // nolint: gomnd
+		return "", "", fmt.Errorf("the bootstrap token %q was not of the form %q", tokenStr, bootstrapapi.BootstrapTokenPattern)
+	}
+	tokenID = substrs[1]
+	tokenSecret = substrs[2]
+
+	return tokenID, tokenSecret, nil
```

Comment:
nit: as the return params are named.

```suggestion
	return substrs[1], substrs[2], nil
```